CSH-10704: Update to trusted publishing

wwjhu opened the pull request:
* Update some of the dependencies, removes eslint-plugin-prettier which slows down linting.
* Update GitHub actions to be able to publish to GitHub Packages

wwjhu committed:
scout: move the newer action versions, add necessary permissions for publishing packages

wwjhu committed:
scout: remove eslint-plugin-prettier.

The main issue: It runs Prettier as an ESLint rule, which means:

Significantly slower linting (Prettier wasn't designed to run this way)
Prettier formatting issues show up as ESLint errors, cluttering your error output with style violations alongside actual code problems
Confusing experience when ESLint reports hundreds of "errors" that are just formatting

Better approach: Run Prettier and ESLint separately:

Use eslint-config-prettier (note: config, not plugin) - this disables ESLint rules that conflict with Prettier
Run Prettier independently (via editor integration, pre-commit hooks, or separate npm scripts)
Let ESLint focus on code quality issues, Prettier handle formatting

wwjhu committed:
chore: update prettier, add prettier-plugin-organize-imports, run prettier:write

wwjhu committed:
scout: add id-token to be able to publish under OIDC (needed due to NPM security updates)

jaapvanblaaderen commented on .github/workflows/release.yml:
```diff
@@ -26,10 +31,11 @@ jobs:
             # https://docs.github.com/en/actions/guides/publishing-nodejs-packages#publishing-packages-to-npm-and-github-packages
             # Setup .npmrc file to publish to GitHub Packages (can replace the other setup-node action when we no longer publish to npm)
-            - uses: actions/setup-node@v4
+            - uses: actions/setup-node@v5
```
Can also be `v6`?

jaapvanblaaderen commented on .github/workflows/release.yml:
I guess specifying the token here should not be needed anymore as this is now covered by the OIDC auth? Also wondering what this action brings instead of just executing `npm publish`

wwjhu replied:
True, but it seems `epeli/npm-release` requires the token to be set. Not sure if we can continue using that action.

jaapvanblaaderen reviewed (commented):
I have some small remarks. Maybe it's also nice to have a few words in the readme on how the OIDC auth is configured (link to: https://docs.npmjs.com/trusted-publishers)?

wwjhu committed:
chore: update action versions

wwjhu commented:
> I have some small remarks. Maybe it's also nice to have a few words in the readme on how the OIDC auth is configured (link to: https://docs.npmjs.com/trusted-publishers)?

That makes sense, will add something

Gecete committed:
CSH-10704: switch npm publishing to OIDC trusted publishing

Gecete committed:
CSH-10704: merge release + prerelease into single publish.yml

npm allows only one trusted publisher per package (matched on workflow filename), so stable and prerelease publishing must share one workflow file. Combine both into publish.yml with event-conditional steps, preserving existing triggers (push to master -> prerelease under `next`, GitHub release -> stable + GitHub Packages).

Gecete committed:
CSH-10704: use Node 24 to drop manual npm upgrade step

Gecete committed:
Merge pull request #200 from WoodWing/csh-10704-trusted-publishing

CSH-10704/trusted publishing continuation

wwjhu committed:
CSH-10704: update readme

wwjhu committed:
CSH-10704: align node versions
